ethpm/backends/registry.py

```diff
@@ -97,8 +100,17 @@ def parse_registry_uri(uri: str) -> RegistryURI:
         raise CannotHandleURI(
             f"Invalid address or ENS domain found in uri: {uri}."
         )
-    parsed_name = parsed_uri.path.strip("/")
-    parsed_version = parsed_uri.query.lstrip("version=").strip("/")
-    pkg_name = parsed_name if parsed_name else None
-    pkg_version = parsed_version if parsed_version else None
+    pkg_name, pkg_version = process_pkg_path(parsed_uri.path)
     return RegistryURI(address, chain_id, pkg_name, pkg_version, ens)
+
+
+def process_pkg_path(pkg_path: str) -> Tuple[str, str]:
+    pkg_id = pkg_path.strip("/")
+    if not pkg_id:
+        return None, None
+
+    if "@" not in pkg_id:
+        return pkg_id, None
```

Comment:
This return and the one above this violate the type signagure of this function....  looks like it should be `Tuple[Optional[str], Optional[str]]`